src/transaction/log_manager.h

```diff
@@ -288,4 +289,32 @@ void LOG_CS_PROMOTE (THREAD_ENTRY * thread_p);
 bool LOG_CS_OWN (THREAD_ENTRY * thread_p);
 bool LOG_CS_OWN_WRITE_MODE (THREAD_ENTRY * thread_p);
 
+//
+// inline implementations
+//
+
+inline bool
+log_is_in_crash_recovery_and_not_yet_completes_redo (void)
+{
+  if (log_Gl.rcv_phase == LOG_RECOVERY_ANALYSIS_PHASE || log_Gl.rcv_phase == LOG_RECOVERY_REDO_PHASE)
+    {
+      return true;
+    }
+  else
+    {
+      return false;
+    }
+}
+
+inline bool
+log_is_in_crash_recovery_but_past_redo_or_restarted (void)
+{
+  if (LOG_RESTARTED == log_Gl.rcv_phase || LOG_RECOVERY_UNDO_PHASE == log_Gl.rcv_phase
```

Comment:
This form, although ugly, is safer for c-style code because of the inherent unsafe nature of the language - avoids accidental assignment (`log_Gl.rcv_phase = LOG_RESTARTED`) which would also evaluate to true.